kolibri/core/assets/test/session-nav-widget.js

```diff
@@ -0,0 +1,110 @@
+              full_name: this.name,
+              username: this.username,
+            },
+          },
+        },
+      });
+      this.vm = new SessionNavWidgetComponent({ store: this.store }).$mount();
+    });
+    describe('initial', function () {
+      beforeEach(function () {
+        this.nameInitial = this.name[0].toUpperCase();
+        this.usernameInitial = this.username[0].toUpperCase();
+      });
+      it('should return \'?\' if user has no name and is not superuser', function () {
+        this.store.state.core.session.full_name = '';
+        assert.equal(this.vm.initial, '?');
+      });
+      it(`should return ${this.usernameInitial} if user has no name and is superuser`, function () {
+        this.store.state.core.session.full_name = '';
+        this.store.state.core.session.kind = [UserKinds.SUPERUSER];
+        assert.equal(this.vm.initial, this.usernameInitial);
+      });
+      it(`should return ${this.nameInitial} if user is a learner`, function () {
+        this.store.state.core.session.kind = [UserKinds.LEARNER];
+        assert.equal(this.vm.initial, this.nameInitial);
+      });
+      it(`should return ${this.nameInitial} if user has name and is a coach`, function () {
+        this.store.state.core.session.kind = [UserKinds.COACH];
+        assert.equal(this.vm.initial, this.nameInitial);
+      });
+      it(`should return ${this.nameInitial} if user has name and is an admin`, function () {
+        this.store.state.core.session.kind = [UserKinds.ADMIN];
+        assert.equal(this.vm.initial, this.nameInitial);
+      });
+      it(`should return ${this.usernameInitial} if user has name and is a superuser`, function () {
+        this.store.state.core.session.kind = [UserKinds.SUPERUSER];
+        assert.equal(this.vm.initial, this.usernameInitial);
+      });
+    });
+    describe('name', function () {
+      it('should return name if user has no kind', function () {
+        assert.equal(this.vm.name, this.name);
+      });
+      it('should return name if user is a learner', function () {
+        this.store.state.core.session.kind = [UserKinds.LEARNER];
+        assert.equal(this.vm.name, this.name);
+      });
+      it('should return name if user is an admin', function () {
+        this.store.state.core.session.kind = [UserKinds.ADMIN];
+        assert.equal(this.vm.name, this.name);
+      });
+      it('should return name if user is a coach', function () {
+        this.store.state.core.session.kind = [UserKinds.COACH];
+        assert.equal(this.vm.name, this.name);
+      });
+      it('should return \'Device Owner\' if user is a superuser', function () {
+        this.store.state.core.session.kind = [UserKinds.SUPERUSER];
+        assert.equal(this.vm.name, this.vm.$tr('deviceOwner'));
+      });
+    });
+    describe('userkind', function () {
+      it('should be Admin if the string is \'Admin\'', function () {
+        if (this.vm.userkind === this.vm.$tr('admin')) {
```

Comment:
I would suggest not having `if` statements here. You should set the state first to ensure this, and then make the assertion. Also, don't worry about testing the `$tr` function, as that is beyond the scope of the tests here.

The best thing to do might be to run a spy on the `$tr` method, once you have set the kind to what you want to test for.
```
var spy = sinon.stub(this.vm, '$tr');
var text = 'admin';
spy.returns(text);
assert.equal(this.vm.userkind, text);
```
and in a separate test with similar preamble:
```
assert.ok(spy.calledWithExactly('admin');
```